### DISCUSSION TITLE
Safari Mouse Wheel Zoom Very Jumpy with Viv Custom Layer

### DISCUSSION THREAD

**Discussion by ilan-gold** [visgl/deck.gl#9189] [Q&A]
Has anyone had any experience where they see on Safari:

https://github.com/user-attachments/assets/5dc6fb76-1512-4943-a7af-1a584f0a3d64

I can provide a repro here: https://codesandbox.io/p/sandbox/reactjs-playground-forked-jy4pkh?file=%2Fpackage.json%3A11%2C31 but it's a bit complicated to provide an up-to-date repro since we are in progress updating to deck.gl `9.x.x`: https://github.com/hms-dbmi/viv/pull/805

This could be related to https://github.com/visgl/deck.gl/issues/6133?

The code for the layer in question is here: https://github.com/hms-dbmi/viv/blob/main/packages/layers/src/image-layer.js and its sublayer: https://github.com/hms-dbmi/viv/tree/main/packages/layers/src/xr-layer

Even help on how to debug this would be amazing!

**Comment by Pessimistress** [comment]
Is this trackpad only? I know that our event handling system treating `wheel` as discrete events, which works reasonably well for mouse wheels. While Macbook trackpad fires `wheel` events in rapid succession.
Unfortunately I do not have the hardware to test a fix...

**Reply by ilan-gold** [reply]
Hello! Yes this is indeed the trackpad although I have not tested elsewhere....I know our office has some mice lying around so I could test.  I am trying to upgrade my OS + Safari now...didn't realize how far back I had fallen. 

Safari: Version 16.6 (18615.3.12.11.2)
macOS: 13.6.1 (22G313)

**Reply by ilan-gold** [reply]
If you're not seeing it, though, then hopefully it's (a) trackpad specific or (b) OS specific

**Reply by ilan-gold** [reply]
I just tried this on someone else's computer with both mouse + trackpad and the same OS/Safari.  Same thing.

**Reply by ilan-gold** [reply]
But macOS 14 is better...not perfect but not as bad.  I will try upgrading soon, but unfortunately our IT can be a little slow.

**Reply by grahambates-code** [reply]
Have you tried using initialViewState instead of viewstate? For some layers viewState is quite janky on OSX for myself.

**Reply by ilan-gold** [reply]
@grahambates-code What exactly are you proposing here? I think we do use `InitialViewState` but are you saying to use it somehow differently than normal?

**Reply by grahambates-code** [reply]
You are using viewState in your code sandbox, not initialViewState.

When you swap to initialViewState it seems a bit better to me.

**Comment by horatiorosa** [comment]
@ilan-gold Did you ever figure out a root cause or solution to this?

**Comment by ilan-gold** [comment]
No the only evidence I gathered was that it was MacOS dependent.  Any insights?
